docs/src/apireference.md

```diff
@@ -547,6 +547,58 @@ Utilities.load
 Utilities.load_constraint
 ```
 
+### Caching optimizer
+
+Some solvers do not support incremental definition of optimization
+models. Nevertheless, you are still able to build incrementally an optimization
+model with such solvers. MathOptInterface allows to use a utility,
+[`Utilities.CachingOptimizer`](@ref), that will store in a [`ModelLike`](@ref) `model_cache`
+the optimization model during its incremental definition. Once the
+model is completely defined, the `CachingOptimizer` specifies all problem's
+information to the underlying solver, all in once.
+
+The function [`Utilities.state`](@ref) allows to query the state
+of the optimizer cached inside a `CachingOptimizer`. The state
+could be:
+* `NO_OPTIMIZER`, if no optimizer is attached;
+* `EMPTY_OPTIMIZER`, if the attached optimizer is empty;
+* `ATTACHED_OPTIMIZER`, if the attached optimizer is synchronized with the
+  cached model defined in `CachingOptimizer`.
+
+Note that there exists methods to modify the attached optimizer:
```

Comment:
I would word this as:
"The following methods modify the state of the attached optimizer:"